identity/generate_test.go

```diff
@@ -45,6 +45,33 @@ var _ = Describe("generate", func() {
 		Expect(identityJson).Should(Equal(orderedJson))
 	})
 
+	It("can upgrade an identity", func() {
+		identity, _ := Create(appConfig, "userID")
+		publicIdentity, _ := GetPublicIdentity(*identity)
+		provIdentity, _ := CreateProvisional(appConfig, "userID@tanker.io")
+		publicProvIdentity, _ := GetPublicIdentity(*provIdentity)
+
+		identityJson, _ := base64.StdEncoding.DecodeString(*identity)
+		publicIdentityJson, _ := base64.StdEncoding.DecodeString(*publicIdentity)
+		provIdentityJson, _ := base64.StdEncoding.DecodeString(*provIdentity)
+		publicProvIdentityJson, _ := base64.StdEncoding.DecodeString(*publicProvIdentity)
+
+		upgradedIdentity, _ := UpgradeIdentity(*identity)
+		upgradedPublicIdentity, _ := UpgradeIdentity(*publicIdentity)
+		upgradedProvIdentity, _ := UpgradeIdentity(*provIdentity)
+		upgradedPublicProvIdentity, _ := UpgradeIdentity(*publicProvIdentity)
+
+		upgradedIdentityJson, _ := base64.StdEncoding.DecodeString(*upgradedIdentity)
+		upgradedPublicIdentityJson, _ := base64.StdEncoding.DecodeString(*upgradedPublicIdentity)
+		upgradedProvIdentityJson, _ := base64.StdEncoding.DecodeString(*upgradedProvIdentity)
+		upgradedPublicProvIdentityJson, _ := base64.StdEncoding.DecodeString(*upgradedPublicProvIdentity)
```

Comment:
You decoded all the base64 only to do a Should(Equal(, you could compare the base64 strings directly instead of the json to simplify this test.